src/operator/subgraph/partition_graph.cc

```diff
@@ -238,21 +239,21 @@ bool LabelSubgraph(const Graph& g,
                          const std::vector<nnvm::Node*>& snodes) {
     if (ancestor == descendant) return true;
     std::stack<const nnvm::Node*> s;
+    std::unordered_set<const nnvm::Node*> visited;
     s.push(descendant);
     size_t count = 0;
     while (!s.empty()) {
-      CHECK_LT(count, indexed_graph.num_nodes()) << "Finding ancestor failed. There is probably"
-                                                    " a loop in the graph";
+      CHECK_LT(count, indexed_graph.num_nodes())
+          << "Finding ancestor failed. There is probably a loop in the graph";
       ++count;
       const nnvm::Node* top = s.top();
       s.pop();
+      visited.insert(top);
       if (top == ancestor) {
         return true;
       }
       for (const auto& entry : top->inputs) {
-        // when searching for the ancestor, the path cannot cross any subgraph node
-        auto it = std::find(snodes.begin(), snodes.end(), entry.node.get());
```

Comment:
Any reason of removing these two lines? Did you forget to initialize `visited` with `snodes`?

Reply:
mmm, missed the snodes feature, for some reason I was thinking snodes was the set of nodes in the stack. Will initialize the visited variable with snodes.

Reply:
are you going to undo the modification?

Reply:
There was originally a bug in the Subgraph API, the change here (with the initialization of visited on line 242 of the new file) fixes that bug as agreed via email with @reminisce.